ecommerce/main.go

```diff
@@ -0,0 +1,105 @@
+package main
+
+import (
+	"fmt"
+	"sort"
+)
+var productName string
+
+type Product struct {
+	name       string
+	price      float64
+	stock      bool
+	stockCount int
+}
+
+func (p *Product) Buy() {
+	if p.stock && p.stockCount > 0 {
+		p.stockCount--
+		fmt.Printf("Purchased %s. New stock count: %d\n", p.name, p.stockCount)
+		if p.stockCount <=0{
+			p.stock = false;
+		}
+	} else {
+		p.stock=false;
+		fmt.Printf("Product %s is out of stock.\n", p.name)
+	}
+}
+
+func searchProduct (products []Product ,name string){
+	for _,product := range products{
+		if product.name == name{
+			fmt.Printf("Your Product %+v",product)
+		}
+	}
+}
+func showOnStockProduct(products []Product){
+	for _,product := range products{
+		if product.stock{
+			fmt.Printf("%+v this product have stock and count of stock %v",product, product.stockCount)
+		}
+		fmt.Printf("%+v this product have not stock",product)
+	}
+}
+func main() {
+	products := []Product{
+		{"Apple", 1.99, true, 20},
+		{"Banana", 0.99, true, 10},
+		{"Orange", 1.49, true, 5},
+	}
+
+	for chosen := true; chosen; {
+		fmt.Println("1. Print All Products")
```

Comment:
Burdaki gibi işlem tiplerine sahip olacağında bunları iota kullanarak constant olarak tutabilirsin. 
ref: https://golangbyexample.com/iota-in-golang/